### PR TITLE
Added an API to access Transport and Firmware Upgrade Logs

### PR DESCRIPTION
**Pull request by dinesharjani**
A new delegate, McuMgrLogDelegate, has been defined, which FirmwareUpgradeManagerDelegate extends, meaning the new protocol is reused to offer a better API for our users.
README file has also been updated to inform library users of this functionality.

### CODE REVIEW

File: Source/McuMgrLogDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//
+//  McuMgrLogDelegate.swift
+//  McuManager
+//
+//  Created by Dinesh Harjani on 16/03/2020.
```
Comment (philips77) [inline Source/McuMgrLogDelegate.swift]: Use the header as in other files.

File: Source/McuMgrLogDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+import Foundation
+
+public protocol McuMgrLogDelegate: class {
+    
+    /// Provides the delegate with content intended to be logged.
+    ///
+    /// - parameter msg: The text to log.
+    func log(_ msg: String)
```
Comment (philips77) [inline Source/McuMgrLogDelegate.swift]: Why the log level type is not returned? This is useful when filtering logs.

**Comment by philips77** [comment]
@bgiori how about making a release on CocoaPods?

**Comment by bgiori** [comment]
@philips77 `0.9.0` released

**Pull request merged**